lib/fastly/client.rb

```diff
@@ -11,9 +11,12 @@ class Client #:nodoc: all
 
     DEFAULT_URL = 'https://api.fastly.com'.freeze
 
-    attr_accessor :api_key, :base_url, :debug, :user, :password, :cookie, :customer
+    attr_accessor :api_key, :base_url, :debug, :user, :password, :cookie, :customer, :without_auth
 
     def initialize(opts)
+      # Some endpoints (POST /tokens) break if any auth headers including cookies are sent so
+      # setting without_auth to true will remove the auth headers in those cases.
+      @without_auth       = opts.fetch(:without_auth, false)
```

Comment:
Instead of setting this global state (which we then need to account for when creating a token a la `incumbent_auth_setting`), could we make it so that specific requests can pass an extra argument that is respected by the `headers` method?

Reply:
I updated the code. Is that what you had in mind?